lsql/judge/views.py

```diff
@@ -396,3 +418,19 @@ def filter_expected_db(expected_db, initial_db):
         if veredict != VeredictCode.AC:
             ret_modified[table] = expected_db[table]
     return ret_added, ret_modified, ret_removed
+
+
+def check_if_get_achievement(user):
+    """Check if the user get some achievement"""
+    num_solved_achievement = NumSolvedAchievementDefinition.objects.all()
+    num_solved_collection = NumSolvedCollectionAchievementDefinition.objects.all()
+    podium_achievement = PodiumAchievementDefinition.objects.all()
+    for num_ach in num_solved_achievement:
+        if not num_ach.check_user(user):
```

Comment:
Mejor haz que `check_and_save(user)` invoque internamente a `check_user(user)` y solo haga trabajo si lo debe hacer, en lugar de hacer la comprobación fuera. 

Para reducirlo aún más, engloba todas los `AchievementDefiniton` en una única lista y recórrela:

    ach_definitions = list(NumSolvedAchievementDefinition.objects.all()) + list(NumSolvedCollectionAchievementDefinition.objects.all()) + ...
    for ach in ach_definitions:
        ach.check_and_save(user)

Reply:
Corregido.